newsroom/history_async.py

```diff
@@ -0,0 +1,178 @@
+import pymongo.errors
+from bson import ObjectId
+import werkzeug.exceptions
+
+from datetime import datetime
+from pydantic import BaseModel
+from typing import Optional, Dict, Annotated, List, Any, Union
+
+from newsroom import MONGO_PREFIX
+from newsroom.users.utils import get_user_or_abort
+from newsroom.utils import get_json_or_400
+from newsroom.core.resources.model import NewshubResourceModel
+from newsroom.core.resources.service import NewshubAsyncResourceService
+from newsroom.users.model import UserResourceModel
+
+from superdesk.core.module import Module
+from quart_babel import gettext
+from superdesk.utc import utcnow
+from superdesk.core.web import EndpointGroup, Response, Request
+from superdesk.flask import abort
+from superdesk.core import json
+from superdesk.core.resources.fields import ObjectId as ObjectIdField
+from eve.utils import ParsedRequest
+from superdesk.core.resources.validators import validate_data_relation_async
+from superdesk.core.resources import (
+    ResourceConfig,
+    MongoResourceConfig,
+    MongoIndexOptions,
+)
+
+
+class RouteParams(BaseModel):
+    item: Optional[Dict[str, Any]] = None
+    action: Optional[str] = None
+    section: Optional[str] = None
+
+
+class HistoryResourceModel(NewshubResourceModel):
+    action: Optional[str] = None
+    versioncreated: Optional[datetime] = None
+    user: Optional[Annotated[ObjectIdField, validate_data_relation_async("users")]] = None
+    company: Optional[Annotated[ObjectIdField, validate_data_relation_async("companies")]] = None
+    item: Optional[str] = None
+    version: Optional[str] = None
+    section: Optional[str] = None
+    extra_data: Optional[Dict] = None
+
+
+class HistoryService(NewshubAsyncResourceService[HistoryResourceModel]):
+    async def create(
```

Comment:
The signature of this function does not match the parent class [signature](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/service.py#L261).

By the looks of it, this is only used by `HistoryService.create_history_record` method. You might be able to simply rename this method from `create` to `create_history_record` (and use `self.create` instead of `super().create` and delete the other `create_history_record`.